### DISCUSSION TITLE
How to get back @AnalyseClasses behavior of Version 0.22.0 for version 1.0.0?

### DISCUSSION THREAD

**Discussion by mikade** [TNG/ArchUnit#975] [Q&A] [answered]
Up to now I am using **Archunit Version with 0.22.0**. I created a bunch of archunit tests for some components, e.g. with package `de.bla.nursing_server`, each arranged as own Maven project in eclipse workspace. Some of these archunit tests are testing against classes from other jars, to prevent the use of classes, e.g. `org.eclipse.jgit.utils.StringUtils`. This `StringUtils` is only available within Eclipse but not within e.g. a Java Application Server. This class belongs to the `org.eclipse.jgit`-Artefakt and finds its way as transitive dependency into the classpath.

```
ArchRule doNotUseEclipseJGit = ArchRuleDefinition.noClasses()
            .should()
            .dependOnClassesThat()
            .haveFullyQualifiedName("org.eclipse.jgit..");
```

The Archunit tests are performed with Junit5,

```
@AnalyzeClasses(packages = { "de.bla.nursing_server" })
public class NursingArchUnitTest {

   @ArchTest
   static final ArchRule doNotUseEclipseJGit = ArchRuleDefinition.noClasses()
                .should()
                .dependOnClassesThat()
                .haveFullyQualifiedName("org.eclipse.jgit..");
```

With **Archunit 0.22.0** the test behaves in a way that classes from `de.bla.nursing_server` package are tested against all classes from the classpath.

Now I want to upgrade to **Archunit 1.0.0** but the test does not work in the previous way.

If I use the current configuration it seems that classes from `de.bla.nursing_server` package are tested against `de.bla.nursing_server` only.

If I use the `wholeClasspath` flag all classes from the classpath are tested against all classes from the classpath.

`@AnalyzeClasses(wholeClasspath = true)`

The same for

`@AnalyzeClasses(packages = { "de.innovas.iaf.nursing_server" }, wholeClasspath = true)`

Therefore my question:

_How should I configure @AnalyzeClasses to get back a behavior like that classes from a selected package are tested against all classes from the classpath?_

**Comment by mikade** [comment, marked as answer]
I am sorry,  i just made an error in building the archunit rule test. 

I used a package regex with Method `ClassesThat.haveFullyQualifiedName()`. This worked with 0.22.0 as desired but not anymore with 1.0.0, for understandable reasons. After replacing `ClassesThat.haveFullyQualifiedName()` with `ClassesThat.resideInAPackage()` all tests were fine.

**Reply by codecholeric** [reply]
Glad you could figure it out!
